transport/src/main/java/io/camunda/zeebe/transport/impl/AtomixServerTransport.java

```diff
@@ -103,7 +104,7 @@ private CompletableFuture<byte[]> handleAtomixRequest(
     final var completableFuture = new CompletableFuture<byte[]>();
     actor.call(
         () -> {
-          final var requestId = requestCount.getAndIncrement();
+          final long requestId = UUID.randomUUID().getMostSignificantBits();
```

Comment:
:x: Getting most significant 64 bit of a 128bit UUID is does not guarantee uniqueness. If we are using UUID we should use the full 128 bits, which in this case is not possible.

I'm thinking of using something like [snowflake id](https://en.wikipedia.org/wiki/Snowflake_ID) which encodes timestamps and nodeid to ensure uniqueness. That said, we don't care about making the id unique over time. It is enough if the id is unique with in a specific period - say a few hours or days. So if we can find something simpler than snowflake id that would be also ok.

Reply:
Agrona has a built-in Snowflake ID generator, which is a single long, so it would be an easy fix.

Reply:
> Agrona has a built-in Snowflake ID generator, which is a single long, so it would be an easy fix.

Great :tada:  Then it would be easy to use it.

Reply:
They have an `IdGenerator` interface which we can use as base type if we don't want to fully commit to Snowflake IDs :smile: